palantir-java-format/src/test/java/com/palantir/javaformat/java/FormatterTest.java

```diff
@@ -452,4 +455,30 @@ void canParse_java9_private_interface_methods() {
                                 + "}"))
                 .doesNotThrowAnyException();
     }
+
+    @Disabled("Disabled as this is proof that PJF always formats the file")
+    @Test
+    void doesntProduceFormattingChangesOnFormattedFiles() throws FormatterException {
+        Formatter formatter = Formatter.create();
+        String simpleClass = "package com.palantir;\n"
+                + "\n"
+                + "public final class Formatted {\n"
+                + "  public static int count(int number) {\n"
+                + "    return number;\n"
+                + "  }\n"
+                + "}";
+        // This is kinda overkill but the point of the test is to show that it
+        // is for sure formatted and still producing a replacement.
+        String formattedClass = formatter.formatSourceAndFixImports(simpleClass);
+        String reformattedClass = formatter.formatSourceAndFixImports(formattedClass);
+
+        // The formatSourceAndFixImports code path flows through getFormatReplacements and
+        // this shows that despite the replacements happening they are superfluous
+        assertThat(formattedClass).isEqualTo(reformattedClass);
+
+        // Proof that the replacements are produced when fully formatted
+        assertThat(formatter.getFormatReplacements(
+                        formattedClass, List.of(Range.closedOpen(0, formattedClass.length()))))
+                .isEmpty();
```

Comment:
should we make the test successful instead ?

```suggestion
        // Proof that the replacements are produced when fully formatted
        List<Replacement> replacements =
                formatter.getFormatReplacements(formattedClass, List.of(Range.closedOpen(0, formattedClass.length())));
        assertThat(replacements.size()).isEqualTo(1);
        assertThat(replacements.getFirst().getReplaceRange()).isEqualTo(Range.closedOpen(0, formattedClass.length()));
```

Reply:
So I considered that, but its kinda weird. Because if it "regresses" that means the underlying bug is fixed... which now that I say it I could put a really clear error message that says revert this change and instead use the API properly.